modules/web/client/src/main/scala/observe/ui/components/MainApp.scala

```diff
@@ -118,8 +121,11 @@ object MainApp:
               confs.find(_.hostName === "*")
         )(orElse = new Exception("Host not found in configuration."))
 
+  // This will run until canceled.
   private def reSync(clientId: ClientId): IO[Unit] =
-    fetchClient.get(RefreshBaseUri / clientId.toString)(_ => IO.unit)
+    fetchClient.get(RefreshBaseUri / clientId.toString)(_ => IO.unit) >>
```

Comment:
You are right, this may be overkill.

Maybe sending only one refresh request is enough. If the WS is broken, we will get the state upon reestablishing anyway.

I'll simplify.

Reply:
I just removed the sleep + infinite loop, so that refresh is only called once.

However, I left the fiber cancellation in place. If the refresh GET fails, it is retried by the middleware. If there's a case where the WS is reestablished and state resynced but the refresh endpoint is still failing, we cancel the fiber.